pipeline/generic_pipeline.go

```diff
@@ -1189,70 +1187,47 @@ func (genericPipeline *GenericPipeline) updateConflictLoggingRules(settings meta
-		// compute the "rules"
-		newRules, err := base.ParseConflictLogRules(conflictLoggingMap)
-		if err != nil || newRules == nil {
-			genericPipeline.logger.Errorf("error converting %v to new conflict logging rules, ignoring the input and continuing with old rules. err=%v",
-				conflictLoggingMap, err)
-			return
-		}
+	// the conflict logger instance will be nil
+	// if conflict logging feature is off right now for this pipeline.
+	conflictLoggingIsOff := genericPipeline.conflictLogger == nil
 
-		// Option 1a: a logger already existed, just update it with the new rules
-		if genericPipeline.conflictLogger != nil {
-			err = genericPipeline.conflictLogger.UpdateRules(newRules)
+	// Option 1: conflict logging needs to be enabled
+	if conflictLoggingEnabled {
+		if conflictLoggingIsOff {
+			// Option 1a: conflict logging needs to be enabled, but it is off right now.
+			newConflictLogger, err := conflictlog.NewLoggerWithRules(conflictLoggingMap, spec.UniqueId(), genericPipeline.logger.LoggerContext(), genericPipeline.logger)
 			if err != nil {
-				genericPipeline.logger.Errorf("error updating existing conflict logging rules to new rules %s, ignoring %v and continuing with old rules. err=%v",
-					newRules, conflictLoggingMap, err)
+				genericPipeline.logger.Errorf("error creating a conflict logger for new rules, ignoring input %v and continuing with old rules. err=%v", conflictLoggingMap, err)
 				return
 			}
-			genericPipeline.logger.Infof("updated conflict logging rules to %s with input %v",
-				newRules, conflictLoggingMap)
+			genericPipeline.conflictLogger = newConflictLogger
 			return
 		}
 
-		// Option 1b: a logger doesn't exist, create a new one with the input rules.
-		clm, err := conflictlog.GetManager()
+		// Option 1b: conflict logging needs to be enabled, but it is on right now.
+		err := conflictlog.UpdateLoggerWithRules(conflictLoggingMap, genericPipeline.conflictLogger, spec.UniqueId(), genericPipeline.logger)
 		if err != nil {
-			genericPipeline.logger.Errorf("error getting conflict logging manager to update to new rules %s, ignoring input %v and continuing with old rules. err=%v",
-				newRules, conflictLoggingMap, err)
-			return
-		}
-
-		spec := genericPipeline.spec.GetReplicationSpec()
-		logger := log.NewLogger(conflictlog.ConflictLoggerName, genericPipeline.logger.LoggerContext())
-
-		newConflictLogger, err := clm.NewLogger(
-			logger,
-			spec.UniqueId(),
-			conflictlog.WithMapper(conflictlog.NewConflictMapper(logger)),
-			conflictlog.WithCapacity(1000), // SUMUKH TODO - make the default size configurable.
-			conflictlog.WithRules(newRules),
-		)
-		if err != nil {
-			genericPipeline.logger.Errorf("error initialising a conflict logger for new rules %s, ignoring input %v and continuing with old rules. err=%v", newRules, conflictLoggingMap, err)
+			genericPipeline.logger.Errorf("error updating existing conflict logging rules, ignoring %v and continuing with old rules. err=%v",
```

Comment:
Should we continue or should the pipeline be restarted where a clean list of rules can be reloaded?
Wouldn't it be odd for customer experience if they edited rules and somehow it's not reflected?